src/metabase/query_processor/util/add_alias_info.clj

```diff
@@ -331,10 +331,14 @@
             (when-let [col (m/find-first (fn [col]
                                            (= (:lib/deduplicated-name col) id-or-name))
                                          source-metadata)]
-              (when-let [desired-column-alias (:lib/desired-column-alias col)]
-                (m/find-first (fn [[_tag _id-or-name opts]]
-                                (= (::desired-alias opts) desired-column-alias))
-                              all-exports)))))
+              (or (when-let [desired-column-alias (:lib/desired-column-alias col)]
+                    (m/find-first (fn [[_tag _id-or-name opts]]
+                                    (= (::desired-alias opts) desired-column-alias))
+                                  all-exports))
+                  (when-let [field-id-or-name (-> col :field_ref second)]
+                    (m/find-first (fn [[_tag id-or-name _opts]]
+                                    (= id-or-name field-id-or-name))
+                                  all-exports))))))
```

Comment:
this is the new fallback.

Reply:
The Field name in `field_ref` in old metadata should be the same as `:name` (which should be the same as `:lib/deduplicated-name`), does it work if you use those instead of `:field_ref`?

Reply:
I'm not sure why `:field_ref` even works at all here since the column should be called `:field-ref` whenever Lib sees it